models/pull.js

```diff
@@ -28,6 +28,18 @@ function Pull(data, signatures, comments, commitStatus, labels) {
       this.data.closes = data.closes;
       this.data.connects = data.connects;
    }
+
+   var isPassing = function() {
+      var repoConfig = config.repos.find(repo => repo.full_name === this.data.repo);
+      var requiredBuilds = repoConfig.required_passing_builds;
+
+      var passingBuilds = this.commitStatuses.filter(
+         build => build.state === 'success');
+
+      return !!_.without(requiredBuilds, ...passingBuilds);
+   };
+
+   this.data.passing = isPassing();
```

Comment:
I don't feel great about this. I'd much rather leave this calculation dynamic than effectively "cache" it as a column in the table.

They each have their advantages. Maybe we should do both? Having it "cached" in the DB certainly makes it easier for analysis and dashboards in SQL. Having it calculated makes it better for dealing with changes in the config, we don't have to clear or recalculate this column.

Oh, this is recalculated each time the pull is loaded from the DB (which is every time it's modified / commented / ...) So that's often enough.